src/devices/blueline.c

```diff
@@ -0,0 +1,330 @@
+ *
+ * While the transmitter ID's are 16-bit, none of them can have any of the two lowest bits set or they
+ * would not be able to transmit their ID as message type 0, and when the offset is calculated (see below) they
+ * would also change the message type.
+ *
+ * For the 3 other message types, the payload must be offset before calculating the CRC or
+ * interpreting the data.  The offset is done by treating the whole payload as a single 16-bit integer and
+ * then subtracting the ID of the transmitter.  After the offset is done, then the CRC may be calculated
+ * and the payload may be interpreted.
+ *
+ * Note that if the transmitter's ID isn't known, the code can't easily determine if messages other than an
+ * ID payload are good or bad, and can't intepret their data correctly.  However, if the BLUELINE_USE_ID_TRANSMISSION
+ * or BLUELINE_USE_ID_AUTODETECT features are enabled, the system can learn the transmitter's ID by various methods.
+ *
+ * For the power message (1), the offset payload gives the number of milliseconds gap between impulses for the most
+ * recent impulses seen by the monitor.  To convert from this 'gap' to kilowatts, you will need your meter's
+ * Kh value. The Kh value is written obviously on the front of most meters, and 1.0 and 7.2 are very common.
+ *
+ * kW = (3600/gap) * Kh
+ *
+ * For the temperature message (2), the offset payload gives the temperature in an odd scaling in the last byte,
+ * and has some flag bits in the first byte.  The only known flag bit is the battery.
+ *
+ * For the energy message (3), the offset payload contains a continuously running power impulse accumulator. I'm
+ * not sure if there is a way to reset the accumulator.  The intended way to use it is to remember the accumulator
+ * value at the beginning of a time period, and then subtract that from the value at the end of the time period.
+ *
+ * kWh = 0.001 * (accumulated pulses) * Kh
+ *
+ * Since the Kh value on all meters can vary, we do not handle it in rtl_433 and just report the raw millisecond
+ * gap and accumulated impulses as received over from the monitor values.
+ *
+ */
+
+// This sets the initial ID the code will try to use.
+// If you know your ID and don't want this code to watch anything else,
+// set that and change both of the following defines to 0.
+#define BLUELINE_DEFAULT_ID 0xB134
+// If BLUELINE_USE_ID_TRANSMISSION is set to 1, this code will jump to monitoring a new ID
+// when someone hits the button on the transmitter.  The code will still REPORT the TXID
+// messages as data even if this is 0, it just won't switch to monitoring that ID.
+// If BLUELINE_ACTIVE_ID_AUTODETECT is set to 1, this code will try to autodetect IDs from
+// incoming messages which otherwise would fail CRC.  While there are a lot of false positives
+// on each individual message, only one ID will continuously match up, and it becomes
+// obvious which one is the right one within a few minutes.
```

Comment:
How many matches do you get? Is there a way to validate the payload parameters?

Reply:
Any individual message will get 2^6 matches.  The payload can unfortunately be any value and there's no way to narrow it down by looking for reserved or unused fields.